integration-cli/docker_cli_daemon_test.go

```diff
@@ -154,6 +155,44 @@ func (s *DockerDaemonSuite) TestDaemonStartIptablesFalse(c *check.C) {
 	}
 }
 
+// Make sure we cannot shrink base device at daemon restart.
+func (s *DockerDaemonSuite) TestDaemonRestartWithInvalidBasesize(c *check.C) {
+	testRequires(c, Devicemapper)
+	c.Assert(s.d.Start(), check.IsNil)
+
+	oldBasesizeBytes := s.d.getBaseDeviceSize(c)
+	newBasesizeBytes := 1073741824 //1GB in bytes
+
+	if int64(newBasesizeBytes) < oldBasesizeBytes {
```

Comment:
The variable above is an `int` and not `int64`. Casting needs to be done for the comparison.
http://play.golang.org/p/EbkRkeGzpV